### PR TITLE
add LB state and healthchecks to task page

### PR DESCRIPTION
**Pull request by benrlodge**
@tpetr 

preview: 
screenshot 2015-03-30 09 47 55

### CODE REVIEW

**Comment by tpetr** [comment]
Looks good overall, just a couple nitpicks:
- might want to rename `Load Balancer State` to `Load Balancer Updates`, since it's table of the updates to the load balancer(s)
- can we consolidate on calling them `Healthchecks` ?
- only show the LB table if `task.taskRequest.loadBalanced` is `true`
- only show the healthchecks table if `task.deploy.healthcheckUri` is a non-empty string
- if `task.deploy.skipHealthchecksOnDeploy` is `true`, display a message like `Healthchecks are disabled for this deploy` with a treatment similar to the `No Health Checks` one.

**Comment by benrlodge** [comment]
Cool - couple quick questions/clarifications

*if task.deploy.skipHealthchecksOnDeploy is true...*
 - I only see this value in the metadata - `deploy.metadata.deploy` as a JSON string - should I parse this, or will there be an actual `task.deploy.healthcheckUri` property if it's true?
screenshot 2015-03-30 11 04 29


*only show the healthchecks table if task.deploy.healthcheckUri is a non-empty string*
 - can I assume this value will always be a string (i.e. can just check the value's .length in hbs)

**Comment by tpetr** [comment]
- yeah, since `healthcheckUri` is an `Optional<String>`, it won't show up in the JSON if it's absent
- can't assume it'd always be a string, but you can coerce it into a boolean using `!!`

**Comment by benrlodge** [comment]
sorry - for that first point, I meant `skipHealthchecksOnDeploy` (not healthcheckUri) doesn't have it's own property but is within a JSON string in `task.taskRequest.deploy.metadata.deploy`

**Comment by tpetr** [comment]
ah sorry -- i misread what you said... yeah, both fields are optional, so the field may not show up in the JSON if they're absent. if `skipHealthchecksOnDeploy` is absent, assume it's `false`

**Comment by benrlodge** [comment]
@tpetr, made these updates.

File: SingularityUI/app/templates/taskDetail/taskHealthChecks.hbs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                                    </a>
+                                </td>
+                            </tr>
+                        {{/each}}
+                    </tbody>
+                </table>
+            </div>
+        {{else}}
+            <div class="empty-table-message">
+                No Health Checks
```
Comment (tpetr) [inline SingularityUI/app/templates/taskDetail/taskHealthChecks.hbs]: could we make this `No Healthchecks` ?

File: SingularityUI/app/templates/taskDetail/taskHealthChecks.hbs [hunk 2]
```diff
@@ -0,0 +1,55 @@
+{{#if synced}}
+    {{#ifCoercedBooleanIsTrue data.task.taskRequest.deploy.healthcheckUri}}
```
Comment (tpetr) [inline SingularityUI/app/templates/taskDetail/taskHealthChecks.hbs]: i think we can get away with just `{{#if data.task.taskRequest.deploy.healthcheckUri}}` here -- is there a situation where it doesnt?

Comment (benrlodge) [inline SingularityUI/app/templates/taskDetail/taskHealthChecks.hbs]: Yea we can get away with that.

**Pull request merged**